src/lib/utils.ts

```diff
@@ -194,22 +193,48 @@ interface ParsedEvent {
   readonly attributes: readonly ParsedAttribute[];
 }
 
+/**
+ * In Tendermint 0.34, event attributes are raw bytes. This changes
```

Comment:
In particular, I believe the ibc_packet data field is binary. Which caused a number of troubles, so they added hex_data field, which was ascii (and I believe what we use). I assume the old binary field is still there.

I would be careful this doesn't throw for protocols outside of ics20 which actually put binary data in the packet data (using the suggestion protobuf, which was NOT used by the ics20 standard for whatever reason - even though they define the format in protobuf 🤷 )

Reply:
Links:
* [Deprecated raw packet data field](https://github.com/cosmos/ibc-go/blob/c86d27fc280cfb342a9e4689b381e5823441b694/modules/core/04-channel/keeper/events.go#L134) is still there, using [packet_data](https://github.com/cosmos/ibc-go/blob/c86d27fc280cfb342a9e4689b381e5823441b694/modules/core/04-channel/types/events.go#L26) key

[bytes for outgoing ica packet](https://github.com/cosmos/ibc-go/blob/c86d27fc280cfb342a9e4689b381e5823441b694/modules/apps/27-interchain-accounts/controller/keeper/relay.go#L66) are ... [still proto-JSON](https://github.com/cosmos/ibc-go/blob/c86d27fc280cfb342a9e4689b381e5823441b694/modules/apps/27-interchain-accounts/types/packet.go#L45-L48) even if they are full of complex protobuf types :shrug:

I guess this is safe enough. I only hit the binary protocol with Desmos, who was following ibc team "best practices" which were never used.

Reply:
Ah, right. Done in 988da83b7977cc40a4b2f2c1478822752e07cf4a.